### DISCUSSION TITLE
Should NVAccess Join the Public Review of the New GitHub Issues Experience?

### DISCUSSION THREAD

**Discussion by christopherpross** [nvaccess/nvda#17255] [General]
Hey NVDA Community,

I wanted to bring up something that could be interesting for us: GitHub has a new public beta for their Issues experience, and I think it might be worth exploring if NVAccess (as an organization) should join in. Given our role as a leader in open-source accessibility, it feels like a great opportunity to provide feedback and help shape this feature. Here’s why I think it’s worth considering:

### What’s New?
1. Issues are now shown in a list format, and the cool part is that you can navigate through them with NVDA using the up and down arrow keys. A simple press of Enter opens the issue you’re on. It’s a bit like having a more intuitive interface for us screen reader users.
   
2. There are some minor bugs with this view, which is exactly why I think our input would be valuable. We could work directly with the developers at GitHub and be part of the conversation to make these features as accessible as possible. I’ve already submitted a bug report here: [GitHub Discussion](https://github.com/orgs/community/discussions/139935?sort=top#discussioncomment-10826666).

3. One of the new features is the ability to create sub-issues. This could be a game-changer, making it easier to break down complex issues into more manageable pieces—kind of like user stories. It might even help us improve the way we organize our backlog.

4. They’re also introducing new issue types, giving us more ways to categorize and handle issues. This could make our work even more efficient.

### How Can We Get Involved?
From what I’ve read, an admin from the NVAccess organization just needs to sign up for the waitlist, and the new experience will be enabled once it’s approved.

### A Little Demo
For anyone interested, here’s a repository where I first saw this new issue view, and I was really excited about it. Maybe you can get a better impression of the new issue layout [here](https://github.com/Bootstrap-Academy/Bootstrap-Academy/issues).

So, what do you all think? Is this something we should try out as a community? It could be a great chance for NVDA to not only benefit from these new features but also to contribute our expertise and ensure they’re accessible for everyone.

Looking forward to hearing your thoughts!

Best,  
Christopher

**Comment by kaveinthran** [comment]
Yes, I really second this initiative and we should join in as the feature
 is being shaped.

Regards,
Kavein
Kaveinthran (He/Him)
Curious, Native Blind

Disabled independent Human Rights Advocate
email: ***@***.*** ***@***.***>
twitter <https://twitter.com/kaveinthran>
My LinkedIn <https://my.linkedin.com/in/kaveinthran>

**Comment by seanbudd** [comment]
I'd be extremely cautious with accessibility issues with this given #15320 stalling for over a year

**Comment by christopherpross** [comment]
@seanbudd 
I completely understand your concerns. I wasn't aware of this issue before. A GitHub employee, who stated they are part of the accessibility team, responded to me in the discussion regarding the accessibility issues. Therefore, I’d like to bring this up for further discussion. If we've already had negative experiences, we should, of course, be cautious when it comes to improving accessibility with GitHub. Personally, I find it extremely frustrating to navigate through the issues on GitHub so far, and I was quite impressed with the new view. Even though these are only minor accessibility issues, I am still looking forward to the view becoming generally available at some point, even if the issues mentioned aren’t resolved by then.

**Reply by seanbudd** [reply]
If the overall experience is more accessible, we should definitely move over. Big hangs up would inaccessible functionality that contributors or NV Access requires.

**Reply by christopherpross** [reply]
I am not aware of any major obstacles. I have tested the feature in another repository and have not noticed any blockages or significant issues in Firefox.

**Reply by CyrilleB79** [reply]
Just tested quickly this new feature.

It's really easier to be able to navigate from issue to issue with the H (heading) shortcut in browse mode!

**Reply by christopherpross** [reply]
@CyrilleB79 
Yeah, and when you are in the focus mode, you can use the arrow keys like in a list to jump from issue to issue. You can use the tab key to get the labels of an issue.

**Reply by seanbudd** [reply]
@christopherpross - have you used this to triage / manage issues as a maintainer?
Has this been tested in Chrome too?

**Reply by christopherpross** [reply]
Hey @seanbudd ,
I have also tested the whole thing under Google Chrome, but did not have the opportunity to test the triage process as a maintainer.

**Comment by SaschaCowley** [comment]
@christopherpross somewhat off topic, but there is an [AxSGrease](https://github.com/jcsteh/axSGrease/) [script for GitHub](https://github.com/jcsteh/axSGrease/blob/master/GitHubA11yFixes.user.js) which helps to alleviate several problems with GitHub's accessibility.

**Reply by christopherpross** [reply]
@SaschaCowley 
Many thanks for the tip.  Unfortunately, the script doesn't give me any headings for the issues, for example, which is actually one of the only things that annoys me about GitHub in collaboration with NVDA. I will leave the script installed anyway.

**Comment by gerald-hartig** [comment]
@christopherpross @CyrilleB79 These new features look very good and I expect we will use them. However this is still a preview, potentially with both bugs and accessibility issues. If the new Issues Experience doesn't work for us, there is no indication that we can revert to the old UI. This change is therefore risky. 

Our past experience with how GitHub has responded to accessibility issues does not fill us with confidence that accessibility issues will be prioritised in a way that matches our needs. We will reconsider this view if someone from the accessibility team at github reaches out and shares their plans on addressing accessibility issues in this feature during the beta period.

That being said, if the new UI is so superior to the old one that it's worth the risk of joining the public-ish beta (they call it a public beta but the wait-list makes it feel more like a private beta), then we can do some further exploration.

**Reply by christopherpross** [reply]
Hey @kaveinthran,
That's understandable. I've already been in contact with a user who, according to him, is a member of the GitHub accessibility team and will take care of the improvements regarding the new issue view. Can I provide more information or be of any other help? Should the discussion remain open for now or should it be closed with this decision?

**Reply by gerald-hartig** [reply]
I'd be happy for the discussion to be closed until we can get some more information on the new system. With the rollout requiring the entire organisation and all its projects to be converted, no indication that we can revert, and a waitlist meaning we can't just set up a dummy project to see how easy it is to do all of the required tasks, we're flying blind.

**Reply by gerald-hartig** [reply]
We can reopen it if the current situation changes or people feel particularly strongly about having the new features.

**Reply by gerald-hartig** [reply]
@christopherpross if you're able to let your contact know, we'd love to see github fix issue https://github.com/orgs/community/discussions/61696 as this would greatly improve accessibility for issue creation.

**Comment by seanbudd** [comment]
Seems like GitHub has rolled this out publicly now

**Comment by SaschaCowley** [comment]
IIRC we can still opt out, but I think the new experience is over all better, despite some issues

**Reply by gerald-hartig** [reply]
What's the accessibility like? Is there any indication we can roll back if it's unusable?

**Comment by SaschaCowley** [comment]
It's ... interesting. There are headings that duplicate information below them. Some parts are overly verbose (the @mentions interface, for example). But keyboard accessibility has been improved, at least in the issues list. And events are now listed separately to comments, which makes things easier to parse (most of the time).

**Reply by owenniblock** [reply]
Hi @SaschaCowley, I'm sorry for the delay replying to you here. I'm one of the engineers on the Issues team (having previously worked on the Accessibility team for a few years.

I had some questions around your feedback above so I can get it properly recorded over here.

> There are headings that duplicate information below them

Could you give me some examples of these? Obviously we don't want to remove any headings that are useful for navigating around the page so I'm keen for us to strike the right balance.

> Some parts are overly verbose

It would be good to get an idea of the worst offenders here, you mentioned the `@mentions` interface, do you mean the experience when you press `@` and start searching for a username?

I'll add internal tracking issues for these and fill in details and clarifications as I get them. I'll also add a feature request for some way to split the comments and events from the timeline.

**Reply by owenniblock** [reply]
@SaschaCowley I just read somewhere else that the grouping of events has improved things, please let me know if the way the events and comments are present is still causing issues and I'll create that tracking issue after I hear from you.

**Reply by SaschaCowley** [reply]
@owenniblock
> > There are headings that duplicate information below them
> 
> Could you give me some examples of these? Obviously we don't want to remove any headings that are useful for navigating around the page so I'm keen for us to strike the right balance.

For instance, when reading through the comments on an issue, this is what is presented (copied from NVDA's Speech Viewer with "Show command keys" turned on):

```
heading    level 2  Activity
down arrow
visited  link    @CyrilleB79's profile
down arrow
heading    level 3  CyrilleB79 commented   2 weeks ago
down arrow
visited  link    CyrilleB79
down arrow
link    2 weeks ago
down arrow
Collaborator
down arrow
menu button  collapsed  subMenu    Comment actions for comment: "Hi @nofarbaranesRG Could you explicitly specify the version of the 
down arrow
menu button  collapsed  subMenu    browser used (Chrome) instead of specifying "latest"... (truncated)"
down arrow
clickable  Hi   link    @nofarbaranesRG
```

Note that the level 3 heading says who commented and when, but this information is then duplicated by the two following links - not to mention the other link for the commenter which precedes the heading.

> you mentioned the @mentions interface, do you mean the experience when you press @ and start searching for a username?

Yep, that's the one. For instance, if I type an @ in this comment field, this is what I hear:

```
at
list
owenniblock Owen Niblock  1 of 5
```

In the new issue experience's comment field, I hear:

```
at
5 autocomplete suggestions available; "seanbudd" is highlighted. Press Enter or Tab to insert. 
list
seanbudd Sean Budd  1 of 5
```

**Reply by owenniblock** [reply]
Great, that's perfect additional context @SaschaCowley - I'll add that information to the issue over here. Thanks!

**Comment by SaschaCowley** [comment]
My bad, looks like events aren't listed separately, I just thought they were because they're in their own region and aren't headings any more. They're still interspersed among the comments.

**Comment by CyrilleB79** [comment]
My first experience of the new issue page, for a few days, is rather negative.

I was used to the old page layout, so a part of things will probably improve once I'll have integrated new ways to navigate.

Thought, I have noted the following drawbacks:
1. Generally, the verbosity is higher and it is somewhat disturbing. E.g. Single button integrated in its own toolbar, label read with their whole description, etc.
2. From the issue list, I have not found the possibility to read a specific issue and turn back in the issue list at the position where I was. Before, for example, if my cursor was on the 3rd issue, I could click the link to open it, and turning back with backspace, I was landing back on the 3rd issue. Or I could press `shift+Enter` to open the issue in a new browser window while keeping the cursor on the 3rd issue. With this new issue list, this does not seem possible anymore.

If some of you have solutions to handle differently these 2 drawbacks, please share it. Else, I'll probably provide a feedback to GitHub.

**Comment by SaschaCowley** [comment]
@CyrilleB79 

> 1. Generally, the verbosity is higher and it is somewhat disturbing. E.g. Single button integrated in its own toolbar, label read with their whole description, etc.

Agree with you completely. I especially dislike the duplication of information in comment headings, and the very verbose list when you \@mention someone.

> 2. From the issue list, I have not found the possibility to read a specific issue and turn back in the issue list at the position where I was. Before, for example, if my cursor was on the 3rd issue, I could click the link to open it, and turning back with backspace, I was landing back on the 3rd issue. Or I could press shift+Enter to open the issue in a new browser window while keeping the cursor on the 3rd issue. With this new issue list, this does not seem possible anymore.

This is really annoying to me too. I've identified the following workarounds, which may or may not work for you:

1. When using browse mode to explore the issues list, and when using focus mode and the arrow keys to explore the issues list, `ctrl+enter` seems to be captured by GitHub and results in it trying to open the issue in a new window. You may need to allow this. In Firefox, you can access the options menu for this alert by pressing `alt+o`, and either allowing pop-ups on GitHub, or allowing specific exceptions through.
2. Weirdly, if you use `j` and `k` to move through the issue list (in focus mode), `ctrl+enter` seems to open the selected issue in a new tab with no problem.
3. You can also use the `applications` key (or `shift+f10`) to open the context menu and open the link in a new tab/window. If you're in focus mode, you'll need to `tab` once to the actual link, as your initial focus is on the entire list item.
4. It doesn't seem like `shift+enter` works on these links at all.

In case it makes a difference, I'm doing this in Firefox.

**Comment by CyrilleB79** [comment]
I have commented in [this discussion](https://github.com/orgs/community/discussions/148713%20and%20press) to provide feedback on the new UX.

Feel free to add more comments or to second what I have reported if necessary.
